OpenRA.Game/Map/CellLayer.cs

```diff
@@ -45,9 +45,23 @@ public void CopyValuesFrom(CellLayer<T> anotherLayer)
 			Array.Copy(anotherLayer.entries, entries, entries.Length);
 		}
 
+		public static CellLayer<T> CreatePrototype(Func<int, int, T> creator, int mapWidth, int mapHeight, TileShape tileShape)
```

Comment:
This will want to be Func<MPos, T> to avoid ambiguity. The function signature is also inconsistent with the CellLayer(TileShape, Size) constructor. Indeed, it seems like you could make this another constructor overload.